File: languages/php/Bloock/Entity/IdentityV2/IntegerEnumAttributeDescriptor.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\IntegerEnumAttributeDefinitionV2;
use Google\Protobuf\Internal\RepeatedField;

/**
 * Represents a descriptor for an attribute with an integer enum value.
 */
class IntegerEnumAttributeDescriptor extends AttributeDescriptor
{
    private RepeatedField $enumeration;

    /**
     * Constructs an IntegerEnumAttributeDescriptor object with the specified parameters.
     * @param string $displayName
     * @param string $technicalName
     * @param string|null $description
     * @param bool $required
     * @param RepeatedField $enumeration
     */
    public function __construct(string $displayName, string $technicalName, ?string $description, bool $required, RepeatedField $enumeration)
    {
        parent::__construct($displayName, $technicalName, $description, $required);
        $this->enumeration = $enumeration;
    }

    public static function fromProto(IntegerEnumAttributeDefinitionV2 $res): IntegerEnumAttributeDescriptor
    {
        return new IntegerEnumAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired(), $res->getEnum());
    }

    public function toProto(): IntegerEnumAttributeDefinitionV2
    {
        $p = new IntegerEnumAttributeDefinitionV2();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        $p->setEnum($this->enumeration);
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/IntegerEnumAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;
import java.util.List;

/**
 * Represents a descriptor for an attribute with an integer enum value.
 */
public class IntegerEnumAttributeDescriptor extends AttributeDescriptor {
  private List<Long> enumeration;

  /**
   * Constructs an IntegerEnumAttributeDescriptor object with the specified
   * parameters.
   * 
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   * @param enumeration
   */
  public IntegerEnumAttributeDescriptor(
      String displayName,
      String technicalName,
      String description,
      Boolean required,
      List<Long> enumeration) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public static IntegerEnumAttributeDescriptor fromProto(
      BloockIdentityEntities.IntegerEnumAttributeDefinition res) {
    return new IntegerEnumAttributeDescriptor(
        res.getDisplayName(),
        res.getId(),
        res.getDescription(),
        res.getRequired(),
        res.getEnumList());
  }

  public BloockIdentityEntities.IntegerEnumAttributeDefinition toProto() {
    return BloockIdentityEntities.IntegerEnumAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .addAllEnum(this.enumeration)
        .build();
  }
}
